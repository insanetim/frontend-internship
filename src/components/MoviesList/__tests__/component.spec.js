import { shallow } from 'enzyme'

import MoviesList from '../component'

it('matches snapshot', () => {
  const mockedMovies = {
    results: [
      {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: 'test/image'
      }
    ],
    total_pages: 10,
    total_results: 200
  }

  const component = shallow(
    <MoviesList
      movies={mockedMovies}
      handlePagination={jest.fn()}
    />
  )

  expect(component).toMatchSnapshot()
})

it('matches snapshot with 1 page', () => {
  const mockedMovies = {
    results: [
      {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: 'test/image'
      }
    ],
    total_pages: 1,
    total_results: 20
  }

  const component = shallow(
    <MoviesList
      movies={mockedMovies}
      handlePagination={jest.fn()}
    />
  )

  expect(component).toMatchSnapshot()
})

it('matches snapshot with no movies', () => {
  const mockedMovies = {
    results: []
  }
  const component = shallow(
    <MoviesList
      movies={mockedMovies}
      handlePagination={jest.fn()}
    />
  )

  expect(component).toMatchSnapshot()
})
