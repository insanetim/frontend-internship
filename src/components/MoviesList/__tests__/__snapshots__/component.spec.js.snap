// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <Row
    gutter={16}
    justify="center"
  >
    <Col
      span={20}
    >
      <Row
        gutter={
          {
            "lg": 32,
            "md": 24,
            "sm": 16,
            "xs": 8,
          }
        }
      >
        <Col
          key="1"
          lg={
            {
              "span": 8,
            }
          }
          md={
            {
              "span": 8,
            }
          }
          sm={
            {
              "span": 12,
            }
          }
          xl={
            {
              "span": 6,
            }
          }
          xs={
            {
              "span": 24,
            }
          }
        >
          <Movie
            actions={[]}
            movie={
              {
                "id": 1,
                "overview": "test/overview",
                "poster_path": "test/image",
                "title": "test/title",
              }
            }
          />
        </Col>
      </Row>
    </Col>
  </Row>
  <Pagination
    onChange={[MockFunction]}
    pageSize={20}
    total={200}
  />
</Fragment>
`;

exports[`matches snapshot with 1 page 1`] = `
<Fragment>
  <Row
    gutter={16}
    justify="center"
  >
    <Col
      span={20}
    >
      <Row
        gutter={
          {
            "lg": 32,
            "md": 24,
            "sm": 16,
            "xs": 8,
          }
        }
      >
        <Col
          key="1"
          lg={
            {
              "span": 8,
            }
          }
          md={
            {
              "span": 8,
            }
          }
          sm={
            {
              "span": 12,
            }
          }
          xl={
            {
              "span": 6,
            }
          }
          xs={
            {
              "span": 24,
            }
          }
        >
          <Movie
            actions={[]}
            movie={
              {
                "id": 1,
                "overview": "test/overview",
                "poster_path": "test/image",
                "title": "test/title",
              }
            }
          />
        </Col>
      </Row>
    </Col>
  </Row>
</Fragment>
`;

exports[`matches snapshot without movies 1`] = `
<Empty
  description="No results"
/>
`;
